README.md

```diff
@@ -2,140 +2,167 @@
+on any infra, rather than build support or guides for each cloud or compute system (e.g. Kubernetes) one by one.
+   * See this in action in 🤗 Hugging Face ([Transformers](https://github.com/huggingface/transformers/blob/main/examples/README.md#running-the-examples-on-remote-hardware-with-auto-setup), [Accelerate](https://github.com/huggingface/accelerate/blob/main/examples/README.md#simple-multi-gpu-hardware-launcher)) and 🦜🔗 [Langchain](https://python.langchain.com/en/latest/modules/models/llms/integrations/runhouse.html)
+
+## 🦾 How does it work?
 
-Runhouse is like **PyTorch + Terraform + Google Drive.**
+In Runhouse, a "cluster" is a unit of compute - somewhere you can send code, data, and requests to execute. It 
+can represent long-lived compute like a static IP or Ray cluster, or ephemeral/scale-to-zero compute like on-demand VMs 
+from a cloud provider or Kubernetes. When you first use a cluster, we check that the hardware is up if applicable (and 
+bring it up if not), and start Ray and a Runhouse HTTP server on it via SSH. Suppose you create a cluster object:
 
-1. Just as **PyTorch** lets you send a model or tensor `.to(device)`, Runhouse OSS
-lets you do `my_fn.to('gcp_a100')` or `my_table.to('s3')`: send functions and data to any of your compute or
-data infra, all in Python, and continue to interact with them eagerly (there's no DAG) from your existing code and
-environment. Think of it as an expansion pack to Python that lets it take detours to remote
-machines or manipulate remote data.
-2. Just as **Terraform** is a unified language for creation and destruction of infra, the
-Runhouse APIs are a unified interface into existing compute and data systems.
-See what we already support and what's on the [roadmap, below](#supported-infra).
-3. Runhouse resources can be shared across environments or teams, providing a **Google Drive**-like
-layer for accessibility, visibility, and management across all your infra and providers.
+```python
+import runhouse as rh
 
-This allows you to:
-* Call your preprocessing, training, and inference each on different hardware from
-inside a single notebook or script
-* Slot that script into a single orchestrator node rather than translate it into an ML pipeline DAG of Docker images
-* Share any of those services or data artifacts with your team instantly, and update them over time
+gpu = rh.cluster(name="my-a100", host=my_cluster_ip, ssh_creds={"user": "ubuntu", "key": "~/.ssh/id_rsa"})
+gpu = rh.cluster(name="my-a100", instance_type="A100:1", provider="cheapest")
+gpu = rh.cluster(name="my-a10", provider="gcp", instance_type="A10:1", zone="us-west1-b", image_id="id-1332353432432", spot=True)
 
-![img.png](https://raw.githubusercontent.com/run-house/runhouse/main/docs/assets/img.png)
-![img.png](https://raw.githubusercontent.com/run-house/runhouse/main/docs/assets/img_1.png)
+gpu.up_if_not()  # Optional, as it's called automatically when you use the cluster
```

Comment:
I also think this line isn't necessary. plus it won't actually work for the first byo cluster instantiated above either.